casper/contracts/simple_casper.v.py

```diff
@@ -472,11 +486,28 @@ def slash(vote_msg_1: bytes <= 1024, vote_msg_2: bytes <= 1024):
         # NO SURROUND VOTE
         slashing_condition_detected = True
     assert slashing_condition_detected
-    # Delete the offending validator, and give a 4% "finder's fee"
+
+    # Slash partial (1/8th deposit of offending validator's deposit, give a 4% "finder's fee"
     validator_deposit: num(wei) = self.get_deposit_size(validator_index_1)
     slashing_bounty: num(wei) = validator_deposit / 25
-    self.total_destroyed += validator_deposit * 24 / 25
-    self.delete_validator(validator_index_1)
+    self.total_destroyed += validator_deposit / 8
+
+    # This is for version 2 to record the malicious validators
+    self.slashed_validators[validator_index_1] = True
+
+    # Slash Total detected, Delete the current offending validator (v1.5)
+    if self.total_destroyed >= floor(self.get_total_curdyn_deposits() * 0.333):
```

Comment:
`total_destroyed` is not a valid check for finding if 1/3 of validators have acted maliciously because only 1/8 of the deposit has been destroyed. It would need to be a check against `total_destroyed * 8`.

That said, `total_destroyed` is global across all epochs so it still doesn't really help us because we only care if 1/3 have acted maliciously in the context of one epoch and actually finalized two conflicting checkpoints. We need to track slashes on a more granular level.

Reply:
How we track slashes on a more granular level is up for debate.

We might be able to throw it in the `votes` struct. We could add in a `slash_bitmap` (for v2) and a `total_slashed` so we then track data specific to an epoch.

Reply:
I think maybe we should make a new array struct `slashes` with a similar structure to `votes` to encapsulate the details of the slashing rather than lumping the data into `votes`.

Open to discussion